File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character,Integer> smap = new HashMap<>();
        Map<Character,Integer> tmap = new HashMap<>();
        for(char c:s.toCharArray()){
            if(!smap.containsKey(c)){
                smap.put(c, 1);
            }else {
                smap.put(c, smap.get(c)+1);
            }
        }
        for(char c:t.toCharArray()){
            if(!tmap.containsKey(c)){
                tmap.put(c, 1);
            }else {
                tmap.put(c, tmap.get(c)+1);
            }
        }
        return smap.equals(tmap);
    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        s_map = {}
        t_map = {}

        for c in s:
            if c not in s_map:
                s_map[c]=0
            s_map[c]+=1

        for c in t:
            if c not in t_map:
                t_map[c]=0
            t_map[c]+=1
        
        if s_map == t_map:
            return True
        else:
            return False